kong/timing/init.lua

```diff
@@ -67,9 +68,17 @@ function _M.auth()
   local http_x_kong_request_debug_token = ngx_var.http_x_kong_request_debug_token
   local http_x_kong_request_debug_log = ngx_var.http_x_kong_request_debug_log
 
-  ngx.req.set_header("X-Kong-Request-Debug", nil)
-  ngx.req.set_header("X-Kong-Request-Debug-Token", nil)
-  ngx.req.set_header("X-Kong-Request-Debug-Log", nil)
+  if http_x_kong_request_debug then
+    ngx_req_set_header("X-Kong-Request-Debug", nil)
+  end
+
+  if http_x_kong_request_debug_token then
+    ngx_req_set_header("X-Kong-Request-Debug-Token", nil)
+  end
+
+  if http_x_kong_request_debug_log then
+    ngx_req_set_header("X-Kong-Request-Debug-Log", nil)
+  end
 
   if http_x_kong_request_debug == nil or
```

Comment:
this fast path `if-return` can be moved to the front

Reply:
We have to make sure that these three headers will not be sent to the upstream, so we have to execute it before the fast path.

Reply:
I understand.